packages/shared/src/chunker.ts

```diff
@@ -0,0 +1,49 @@
+// No clue why, but 3600 matches 4kb in the browser
+const MAX_CHUNK_SIZE = 3600;
+
+interface Chunk {
+	name: string;
+	value: string;
+}
+
+// Get cookie as chunks to set in the browser
+export function createChunks(key: string, value: string) {
+	// check the length and then split
+	const chunkCount = Math.ceil(value.length / MAX_CHUNK_SIZE);
+
+	if (chunkCount === 1) {
+		return value;
+	}
+
+	const chunks: Chunk[] = [];
+	const values = value.match(new RegExp('.{1,' + MAX_CHUNK_SIZE + '}', 'g'));
```

Comment:
Move the regexp to a constant, as creating a new regexp every time is very slow.